Jenkinsfile

```diff
@@ -92,56 +92,149 @@ pipeline {
-                  echo "$(retrieve_token "full" "invalid_audience")" > "gcp_token_invalid_audience"
-                  echo "$(retrieve_token "standard" "conjur/cucumber/host/test-app")" > "gcp_token_standard_format"
+                ./get_gce_tokens_to_files.sh || exit 1
                 '''
+                stash name: 'authnGceTokens', includes: 'gcp_token_*', allowEmpty:false
+              }
+            }
+          }
+          post {
+           failure {
+            script {
+              env.GCP_ENV_ERROR = "true"
+            }
+           }
+           success {
+            script {
+              env.GCE_TOKENS_FETCHED = "true"
+            }
+            echo '-- Finished fetching GCE tokens.'
+           }
+          }
+        }
+        /**
+        * We have 3 stages for GCP Authenticator tests. The first one runs inside
+        * In this stage, Google SDK container executes a script to deploy a function,
+        * the function accepts audience in query string and returns a token with that audience.
+        * All the tokens required for testings are obtained and written to function directory, the post stage branch
+        * deletes the function.
+        * This stage depends on stage: 'GCP - Stage A - Compute Engine' to get the GCP project.
+        */
+        stage('GCP - Stage B - Google Function') {
+          environment {
+            GCP_FETCH_TOKEN_FUNCTION = "fetch_token_${BUILD_NUMBER}"
+            IDENTITY_TOKEN_FILE = 'identity-token'
+            GCP_OWNER_SERVICE_KEY_FILE = "sa-key-file.json"
+            GCP_ZONE="us-central1"
+          }
+          steps {
+            echo "Waiting for GCP project name"
+            timeout(time: 10, unit: 'MINUTES') {
+              waitUntil {
+                script {
+                  return (env.GCP_PROJECT != null || env.GCP_ENV_ERROR == "true")
+                }
+              }
+            }
+            script {
+              if (env.GCP_ENV_ERROR == "true") {
+                error('GCP_ENV_ERROR cannot deploy function')
+              }
 
-                stash name: 'authnGcpTokens', includes: 'gcp_token_valid,gcp_token_invalid_audience,gcp_token_standard_format,gcp_token_user,gcp_token_non_existing_host,gcp_token_non_existing_account,gcp_token_non_rooted_host', allowEmpty:false
-                env.GCP_TOKENS_FETCHED = "true"
+              dir('ci/authn-gcp') {
+                sh '''
+                # Deploy Google Cloud function
+                summon ./run_gcloud.sh deploy_function.sh || exit 1
+                if [ $? -ne 0 ]; then
+                    echo '-- Error deploying Google function'
+                    exit 1
+                fi
+                # Obtain tokens from Google function and write to files
+                ./get_func_tokens_to_files.sh
+                if [ $? -ne 0 ]; then
+                    echo '-- Error obtaining tokens from Google function'
+                    exit 1
+                fi
+                '''
```

Comment:
we moved the other script from Jenkinsfile to a separate file. Can we do the same here?